### PR TITLE
[14615] Add tests for DomainParticipant::find_topic

### PR DESCRIPTION
**Pull request by MiguelCompany**
## Description

Following TDD, this PR adds tests for `DomainParticipant::find_topic` before implementing it

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- [x] Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added. 
- **N/A** Any new/modified methods have been properly documented using Doxygen. 
- [x] Fast DDS test suite has been run locally. 
       Checked that the new tests fail.
- [x] Changes are ABI compatible. 
- [x] Changes are API compatible. 
- **N/A** Documentation builds and tests pass locally. 
- **N/A** New feature has been added to the `versions.md` file (if applicable).
- **N/A** New feature has been documented/Current behavior is correctly described in the documentation. 

## Reviewer Checklist
- [ ] Check contributor checklist is correct.
- [ ] Check CI results: changes do not issue any warning.
- [ ] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: test/blackbox/common/DDSBlackboxTestsFindTopic.cpp [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+namespace eprosima {
+namespace fastdds {
+namespace dds {
+
+using ReturnCode_t = eprosima::fastrtps::types::ReturnCode_t;
+
+class DDSFindTopicTest : public testing::Test
+{
+    /**
+     * A mock type support class.
```
Comment (EduPonz) [inline test/blackbox/common/DDSBlackboxTestsFindTopic.cpp:50]: I think this is a dummy
```suggestion
     * A dummy type support class.
```

File: test/blackbox/common/DDSBlackboxTestsFindTopic.cpp [hunk 2]
```diff
@@ -0,0 +1,386 @@
...
+    }
+
+};
+
+/**
+ * Implements test DDS-DP-FT-01 from the test plan.
+ *
+ * Check timeout behavior of DomainParticipant::find_topic.
+ */
+TEST_F(DDSFindTopicTest, find_topic_timeout)
```
Comment (EduPonz) [inline test/blackbox/common/DDSBlackboxTestsFindTopic.cpp:156]: I think we can make this test way faster

Comment (MiguelCompany) [reply test/blackbox/common/DDSBlackboxTestsFindTopic.cpp:156]: I think it should be at least one scheduler quantum. How about 50 ms?

**Review by EduPonz — APPROVED** [review]
LGTM

**Pull request merged**